### ISSUE TITLE
Simple way to see SQL statements

### ISSUE THREAD

**Issue by gzinger** [aspnet/EntityFramework.Docs#1331]
In EF6 it was very simple to output the generated SQL with a one-line of code using Database.Logging property.
I tried to do everything discussed in this article as well as ASP.NET Core logging linked article, but it doesn't work.
Could you please provide a COMPLETE sample of how to see the EF generated SQL in a ASP.NET Core Web API?
Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: ed83069f-39a8-e992-95f7-a8b41154b6f4
* Version Independent ID: 144fe168-080f-6557-e4b2-a18adb4b89ed
* Content: [Logging - EF Core](https://docs.microsoft.com/en-us/ef/core/miscellaneous/logging)
* Content Source: [entity-framework/core/miscellaneous/logging.md](https://github.com/aspnet/EntityFramework.Docs/blob/master/entity-framework/core/miscellaneous/logging.md)
* Product: **entity-framework**
* GitHub Login: @rowanmiller
* Microsoft Alias: **divega**

**Comment by jayeshdshah** [comment]
Pl. check this article, may it help you...

http://thedatafarm.com/data-access/logging-in-ef-core-2-2-has-a-simpler-syntax-more-like-asp-net-core/

**Comment by kwaazaar** [comment]
It already writes every statement to the logs with loglevel Information.

**Comment by jemiller0** [comment]
Yes, for the love of god, please simply make it work like it did in EF 6. I thought you were going to do this a long time ago? I would swear this was on the list years ago. The ASP.NET Core logging infrastructure is an absolute train wreck. Not everyone is using this with ASP.NET Core. I had to jump through numerous hoops to get this working in an earlier version of EF Core and now that I have upgraded from EF Core 2.2 to 3.1, it no longer works and I see that the documentation on how to get it to work has changed. This should not be this complicated. Apparently, something on the life cycle of the logging provider changed or something. It works for me on the first instance of the DbContext. Then, when it goes out of scope, logging no longer works for additional instances. The way I've had it working up to this point was to add my provider using the following code in OnModelCreating().

            this.GetService<ILoggerFactory>().AddProvider(new LoggingProvider());

I'm assuming it must have been storing it in a static variable behinds the scene in previous versions of EF and now it isn't.

**Comment by ErikEJ** [comment]
This is implemented in EF Core 5! (Even better than in EF6)

**Comment by ajcvickers** [comment]
See https://github.com/dotnet/efcore/issues/19549#issuecomment-572823246
Also document CreateDbCommand: https://github.com/dotnet/efcore/issues/19358

**Comment by StevenRasmussen** [comment]
Might be worth updating this document to include a note that if you add 1 line to the `LogLevel` node appsettings.json that you will get EF Core logging:
`"Microsoft.EntityFrameworkCore.Database.Command": "Information"`

Here is the full path to the setting:
```
{
  "Logging": {
    "LogLevel": {
      "Microsoft.EntityFrameworkCore.Database.Command": "Information"
    }
  }
}
```

**Comment by Rick-Anderson** [comment]
@StevenRasmussen  has the right answer. I'll try to get this updated in the next quarter.
See [this SO thread](https://stackoverflow.com/questions/45893732/how-do-you-show-underlying-sql-query-in-ef-core) with 47K views. [Several other SO threads](https://www.google.com/search?q=entity+framework+core+show+sql+in+output+window&rlz=1C1CHBF_enUS801US801&oq=entity+framework+core+show+SQL+output&aqs=chrome.1.69i57j0i22i30.11816j0j7&sourceid=chrome&ie=UTF-8)  on the same question. I was unable to find an official doc on this. See [my Answer](https://stackoverflow.com/a/66662556/502537)

**Comment by ajcvickers** [comment]
Note: update https://aka.ms/efcore-docs-query-string when this is done.

**Comment by jemiller0** [comment]
The link to the documentation is broken. In any case, I switched to Dapper since you threw .NET Framework under the bus.

**Comment by ajcvickers** [comment]
@jemiller0 The link is in the process of being created.
